mkosi/__init__.py

```diff
@@ -1585,6 +1585,25 @@ def find_devicetree(context: Context, kver: str) -> Path:
     die(f"Requested devicetree {context.config.devicetree} not found")
 
 
+def find_devicetrees_auto(context: Context, kver: str, directory: Optional[Path] = None) -> list[Path]:
```

Comment:
Why have a `directory` argument here? Just use `context.config.devicetree_auto_directory` directly in this function.